src/main/java/dev/sigstore/rekor/client/HashedRekordRequest.java

```diff
@@ -57,20 +58,35 @@ public static HashedRekordRequest newHashedRekordRequest(
                 new Signature()
                     .withContent(Base64.getEncoder().encodeToString(signature))
                     .withPublicKey(
-                        new PublicKey().withContent(Base64.getEncoder().encodeToString(certPem))));
-    return new HashedRekordRequest(hashedrekord);
+                        new PublicKey()
+                            .withContent(Base64.getEncoder().encodeToString(publicKey)))));
   }
 
+  /** Returned a canonicalized json payload. */
   public String toJsonPayload() {
+    // TODO: use RekorEntryBody type here
     var data = new HashMap<String, Object>();
     data.put("kind", "hashedrekord");
     data.put("apiVersion", "0.0.1");
-    data.put("spec", hashedrekord);
+    data.put("spec", hashedRekord);
 
-    return GSON.get().toJson(data);
+    try {
+      return new JsonCanonicalizer(GSON.get().toJson(data)).getEncodedString();
+    } catch (IOException ioe) {
+      // we shouldn't be here
+      throw new RuntimeException(
+          "GSON generated invalid json when serializing HashedRekordRequest");
+    }
   }
 
   public HashedRekord getHashedRekord() {
-    return hashedrekord;
+    return hashedRekord;
+  }
+
+  /** Computes the expected rekor uuid of an entry based on the content of the hashedRekord. */
+  public String computeUUID() {
```

Comment:
It's a bit surprising that the UUID is actually just a hash.

Reply:
Yeah it's how it works in trillian.